vendor/pischem/caddy/src/etc/rc.d/caddy

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
```

Comment:
Is caddy not shipping an appropriate rc file?

Reply:
I didn't find one. Also this is an integral part for the working plugin since it also executes a setup script before each caddy start that triggers certain functions like certificates being copied, the Caddyfile being reformatted and making sure the whole folder structure is clean etc...

If this can't be used I don't know an alternative right now.

Reply:
# pkg info -l caddy-custom | grep rc.d
	/usr/local/etc/rc.d/caddy

;)

Reply:
It seems like there's a pretty big rc.d file with lots of functionality included. Sadly it misses functionality that I need like calling my custom setup script. I don't know how to proceed here. I need help.

Reply:
Rename this file otherwise it caused a package conflict.

Reply:
For simplicity maybe „opncaddy“

Reply:
What exactly in your rc is not supported by the original one?

Reply:
> What exactly in your rc is not supported by the original one?

I'm triggering a custom setup script that does a few things. I have a few more folders than in the standard rc.d script.

https://github.com/Monviech/opnsense-plugins/blob/f57adeafc545af4e8c6971ef50fef862c986d6af/www/caddy/src/opnsense/scripts/Pischem/Caddy/setup.sh

Exactly before each caddy start (so after the template has generated the Caddyfile) I have to trigger a "caddy fmt --overwrite" to fix all formatting errors that occur when generation from jinja2 template to Caddyfile occured. It's almost impossible to get the file exactly right without it. Caddy still starts, would print an warning, but the Caddyfile would look horrible.

https://github.com/Monviech/opnsense-plugins/blob/f57adeafc545af4e8c6971ef50fef862c986d6af/www/caddy/src/opnsense/scripts/Pischem/Caddy/setup.sh#L40

Also I hooked certificate export from the OPNsense config.xml into that script too. It exports all certificates, puts them into a temp folder, and caddy loads and deletes them automatically when it starts.

https://github.com/Monviech/opnsense-plugins/blob/f57adeafc545af4e8c6971ef50fef862c986d6af/www/caddy/src/opnsense/scripts/Pischem/Caddy/setup.sh#L43

If the functionality of this custom folder structure, the exact moment of the caddy reformatting, and the certificate export can all be done at a different spot, I'm happy to use the standard rc.d file in which I can't call the custom setup script.

Reply:
Huh, I got the setup stuff into FreeBSD even. A lot of core and plugins use it without rc modifications.

Reply:
https://github.com/opnsense/plugins/blob/2c943b4de4d5afaab83172e558479c632ee6dd4d/net/frr/src/opnsense/service/templates/OPNsense/Quagga/frr#L4

Reply:
I didn't know something like this is possible. I'm going to look into it. I want to prevent the custom rc.d if possible.

Reply:
I get the feeling the last one is overcomplicating the simple solution of adding caddy_setup="/path/to/setup.sh" by keeping the stub, renaming everything and then still using the original package script. if something goes wrong this will be hard to debug and won't benefit from core changes on the subject.

Reply:
I have created an opncaddy wrapper script. The current caddy-custom rc.d script can be used unaltered, the plugin just uses the opncaddy script so it can inject the setup.sh script into the start/stop/restart actions, after the setup script has been invoked it lets the normal caddy-custom rc.d script continue with all of its processing.

https://github.com/Monviech/opnsense-plugins/commit/70b963a09d46af4eff39b8d299f003d5a53df220

Reply:
Reverted this change, different less intrusive way with rc.conf is evaluated.

Reply:
Works as expected in rc.conf, change pushed and issue resolved.